### ISSUE TITLE
Update `dthcaus_source()` and `lstalvdt_source()`

### ISSUE THREAD

**Issue by thomas-neitmann** [Roche-GSK/admiral#583]
The first argument `dataset` which currently expects a data frame should be renamed to `dataset_name` and accept the name of a dataset, e.g. `"adsl"`, as a string. The functions which perform the actual logic, i.e. `derive_var_dthcaus()` and `derive_var_lstalvdt()`, should get a `source_datasets` parameter which expects a list of data frames that corresponds to the names specified in `dataset_name`.

**Comment by bundfussr** [comment]
I wanted to implement it in a separate pull request once https://github.com/Roche-GSK/admiral/issues/546 is done. I have implemented an assertion function for checking the list of sources and the `source_datasets` parameter in https://github.com/Roche-GSK/admiral/issues/546.
It makes sense to do it after the deprecation logic is available but I am not sure if it should be done within the same pull request.

**Comment by bundfussr** [comment]
I just noticed that the Changelog was not updated for this issue. @thomas-neitmann , could you update it?
